### PR TITLE
update catalog api

### PR DESCRIPTION
**Pull request by yqguodataman**

### CODE REVIEW

File: src/plugins/catalog/model.go [hunk 1]
```diff
@@ -2,7 +2,7 @@ package catalog
 
 import (
 	"encoding/base64"
-	"errors"
+	//"errors"
```
Comment (upccup) [inline src/plugins/catalog/model.go]: unuse code

File: src/plugins/catalog/store.go [hunk 2]
```diff
@@ -41,3 +45,21 @@ func (catalogApi *CatalogApi) Get(catalogId uint64) (Catalog, error) {
...
+		Where("name = ? AND id != ? AND user_id = ?", catalog.Name, catalog.ID, catalog.UserId).
+		Count(&count).
+		Error; err != nil {
+		log.Errorf("get catalog error: %v", err)
+		return err
+	}
+
+	if count > 0 {
+		log.Warnf("catlog: %s already exist", catalog.Name)
+		return errors.New("already exist")
```
Comment (upccup) [inline src/plugins/catalog/store.go]: crane error with error code

File: src/plugins/catalog/store.go [hunk 3]
```diff
@@ -41,3 +45,21 @@ func (catalogApi *CatalogApi) Get(catalogId uint64) (Catalog, error) {
 func (catalogApi *CatalogApi) Delete(catalogId uint64) error {
 	return catalogApi.DbClient.Delete(&Catalog{ID: catalogId}).Error
 }
+
+func (catalogApi *CatalogApi) Update(catalog *Catalog) error {
+	count := 0
+	if err := catalogApi.DbClient.
+		Model(&Catalog{}).
+		Where("name = ? AND id != ? AND user_id = ?", catalog.Name, catalog.ID, catalog.UserId).
```
Comment (cmingxu) [inline src/plugins/catalog/store.go]: id != ? AND user_id = ? AND name = ?  might give MySQL hint for better performance.

File: src/plugins/catalog/store.go [hunk 4]
```diff
@@ -41,3 +45,21 @@ func (catalogApi *CatalogApi) Get(catalogId uint64) (Catalog, error) {
 func (catalogApi *CatalogApi) Delete(catalogId uint64) error {
 	return catalogApi.DbClient.Delete(&Catalog{ID: catalogId}).Error
 }
+
+func (catalogApi *CatalogApi) Update(catalog *Catalog) error {
+	count := 0
+	if err := catalogApi.DbClient.
```
Comment (cmingxu) [inline src/plugins/catalog/store.go]: how about extract to a function with code line#10 - line#15, 
catalogCountWithName(catalogName)

File: src/plugins/catalog/api.go [hunk 5]
```diff
@@ -89,7 +100,12 @@ func (catalogApi *CatalogApi) CreateCatalog(ctx *gin.Context) {
 		return
 	}
 
-	catalogApi.Save(&catalog)
+	catalog.UserId = account.(auth.Account).ID
```
Comment (cmingxu) [inline src/plugins/catalog/api.go]: UserId  => AccountId  would be great as we do not have such model named "User", we use "Account"

File: deploy/mysql/init.sql [hunk 6]
```diff
@@ -31,9 +31,10 @@ CREATE TABLE `catalogs` (
 	  `bundle` longtext,
 	  `description` longtext,
 	  `icon_data` longtext,
-	  `user_id` bigint(20) unsigned DEFAULT NULL,
+	  `account_id` bigint(20) unsigned DEFAULT NULL,
 	  `type` int(10) unsigned DEFAULT NULL,
-	  PRIMARY KEY (`id`)
+	  PRIMARY KEY (`id`),
+	  UNIQUE KEY `accountid_name` (`account_id`, `name`)
```
Comment (vitan) [inline deploy/mysql/init.sql]: We have to promise the version upgrade by DB migration scripts from the coming version

File: src/plugins/catalog/store.go [hunk 7]
```diff
@@ -41,3 +19,7 @@ func (catalogApi *CatalogApi) Get(catalogId uint64) (Catalog, error) {
 func (catalogApi *CatalogApi) Delete(catalogId uint64) error {
 	return catalogApi.DbClient.Delete(&Catalog{ID: catalogId}).Error
 }
+
+func (catalogApi *CatalogApi) Update(catalog *Catalog) error {
+	return catalogApi.DbClient.Save(catalog).Error
```
Comment (vitan) [inline src/plugins/catalog/store.go]: 这个需要 unittest

**Comment by vitan** [comment]
resolving #404

**Pull request merged**